protected/controllers/PracticeController.php

```diff
@@ -2,62 +2,179 @@
+                'users' => array('*'),
             ),
         );
     }
-    public function actionCreate()
+    /**
+     * Displays a particular model.
+     * @param integer $id the ID of the model to be displayed
+     */
+    public function actionView($id)
+    {
+        $this->render('view', array(
+            'model' => $this->loadModel($id),
+        ));
+    }
+
+    public function actionCreate($context=null)
     {
         $contact = new Contact();
         $address = new Address();
         $practice = new Practice();
 
-        $contact->attributes = $_POST['Contact'];
-        $address->attributes = $_POST['Address'];
-        $practice->attributes = $_POST['Practice'];
-        list($contact,$practice,$address) = $this->performPracticeSave($contact, $practice,$address);
+        $this->performAjaxValidation($contact);
 
-        echo CJSON::encode(array('label'=> $practice->getAddressLines()
-        ,'value'=> $practice->getPrimaryKey()));
+        if (isset($_POST['Contact'])) {
+            $contact->attributes = $_POST['Contact'];
+            $address->attributes = $_POST['Address'];
+            $practice->attributes = $_POST['Practice'];
+            list($contact, $practice, $address) = $this->performPracticeSave($contact, $practice, $address, $context === 'AJAX');
+        }
+        if ($context === 'AJAX') {
+            echo CJSON::encode(array('label'=> $practice->getAddressLines()
+            ,'value'=> $practice->getPrimaryKey()));
+        } else {
+            $this->render('create', array(
+                'model' => $practice,'address' => $address, 'contact' => $contact,
+            ));
+        }
     }
 
-    public function performPracticeSave(Contact $contact, Practice $practice,Address $address)
+    public function performPracticeSave(Contact $contact, Practice $practice,Address $address,$isAjax = false)
     {
         $transaction = Yii::app()->db->beginTransaction();
         try {
-            if( $contact->save() ){
+
+            if( $contact->save()){
                 $practice->contact_id = $contact->getPrimaryKey();
                 $address->contact_id = $contact->id;
-                if($practice->save()  && $address->save()) {
-                    $transaction->commit();
-
+                if($practice->save()) {
+                    if (isset($address)) {
+                        if ($address->save())
+                            $transaction->commit();
+                        else
+                            $transaction->rollback();
+                    }
+                    else{
+                        $transaction->commit();
+                    }
+                    if (!$isAjax) {
+                        $this->redirect(array('view', 'id' => $practice->id));
+                    }
                 }
                 else {
                     $transaction->rollback();
                 }
             }
             else {
+                var_dump($contact->getErrors());
```

Comment:
This var_dump should be removed. Any errors should already appear onscreen due to AJAX validation.